src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJLibrary.java

```diff
@@ -58,4 +86,10 @@ public Path getPath() {
 
   @CalledWithWriteLock
   protected abstract void initializeLibraryProperties(LibraryProperties<S> properties);
+
+  @Override
+  @Nullable
+  public com.intellij.openapi.roots.libraries.Library getPlatformObject() {
```

Comment:
this probably could have a more explaining name, like `get...Library()`?, as it returns a `Library` and that's what's happening inside the method too.
also the part that is `...Platform...` why is it platform, what is sematical link between it and the `IntelliJLibrary` class?

Reply:
The point here is that all `IntelliJComponent<T>` objects have some "platform object" (of type `T`) that is their correspondent in the IntelliJ platform's object hierarchy.

IntelliJLibrary is a component of ours. Its existence is based in the in information in the course configuration file (hypothetically, as we haven't really implemented those kind of libraries yet, but they are kind of supported by our design) or information "hard-coded" to our plugin ("common libraries" as, and currently only, Scala SDK). IntelliJLibrary knows where to download its binaries and where to store them and such things.

The platform object in this case is a `com.intellij.openapi.roots.libraries.Library` object. It's an object created and managed by IntelliJ platform and its existence is tied to that specific library being loaded in the project.

So if our IntelliJModule is in LOADED state, there should be a "platform object" for it. That object is returned by this method.

Hope this makes sense.